src/auth/mod.rs

```diff
@@ -127,17 +117,18 @@ impl FromRequest<ServerState> for HawkPayload {
             request.uri().path_and_query().ok_or(AuthError)?.as_str(),
             request.uri().host().unwrap_or("127.0.0.1"),
             request.uri().port().unwrap_or(settings.port),
-            &request.state().master_token_secret,
+            &request.state().secrets,
             Utc::now().timestamp() as u64,
         )
     }
 }
 
-fn hkdf_expand_32(info: &[u8], salt: Option<&[u8]>, key: &[u8]) -> AuthResult<[u8; 32]> {
+pub fn hkdf_expand_32(info: &[u8], salt: Option<&[u8]>, key: &[u8]) -> [u8; 32] {
     let mut result = [0u8; 32];
     let hkdf: Hkdf<Sha256> = Hkdf::extract(salt, key);
-    hkdf.expand(info, &mut result)?;
-    Ok(result)
+    // This unwrap will never panic because 32 bytes is a valid size for Hkdf<Sha256>
+    hkdf.expand(info, &mut result).unwrap();
```

Comment:
Potentially contentious change alert: This is 100% safe and will not fail as long as `result` is a `[u8;32]`. But also it's an `unwrap` in production code and that's not cool, I know.

I was hoping to avoid having to propagate the error type out through the call stack, because this function is called more widely (anywhere that calls `Setting::new` would need to implement error handling, even though we know the error won't happen in practice).